### PR TITLE
[casr-ubsan] Fix crashline parsing

### PR DESCRIPTION
**Pull request by hkctkuy**

### CODE REVIEW

**Review by SweetVishnya — CHANGES_REQUESTED** [review]
We need a test for the new case

File: libcasr/src/ubsan.rs [hunk 1]
```diff
@@ -72,65 +68,99 @@ impl Severity for UbsanWarning {
...
+            let Ok(header_crashline) = header_crashline else {
+                // Return error as it is
+                return header_crashline;
+            };
+            let CrashLine::Source(stack_crashline) = stack_crashline else {
+                // header_crashline may be CrashLine::Source
+                return Ok(header_crashline);
+            };
+            let CrashLine::Source(header_crashline) = header_crashline else {
+                return Ok(header_crashline);
```
Comment (Avgor46) [inline libcasr/src/ubsan.rs:143]: Why do we return `header_crashline`? We've parsed `stack_crashline` with sources.

File: libcasr/src/ubsan.rs [hunk 2]
```diff
@@ -72,65 +68,99 @@ impl Severity for UbsanWarning {
...
-                Ok(CrashLine::Source(crashline.debug))
-            } else if !crashline.module.is_empty() && crashline.offset != 0 {
-                Ok(CrashLine::Module {
-                    file: crashline.module,
-                    offset: crashline.offset,
-                })
+            let Ok(stack_crashline) = get_crash_line(crashline) else {
+                return header_crashline;
+            };
+            let Ok(header_crashline) = header_crashline else {
```
[the remark this answers is not in the document]
Comment (hkctkuy) [reply libcasr/src/ubsan.rs:134]: Done

File: libcasr/src/ubsan.rs (hunk 1 above)
Comment (hkctkuy) [reply libcasr/src/ubsan.rs]: Done

**Review by Avgor46 — APPROVED** [review]

File: libcasr/src/ubsan.rs [hunk 3]
```diff
@@ -72,65 +68,98 @@ impl Severity for UbsanWarning {
...
+
         let message = self.ubsan_report();
         if message.is_empty() {
             return Err(Error::Casr("Empty ubsan message".to_string()));
         }
-        // If there is no stacktrace use crashline from first string
-        // May be not absolute
-        // Else use first string from stacktrace
-        let crashline = if let Some(crashline) = message
+        // If there is no stacktrace use first string from stacktrace
```
Comment (SweetVishnya) [inline libcasr/src/ubsan.rs:122]: If there is a stack trace

File: libcasr/src/ubsan.rs [hunk 4]
```diff
@@ -72,65 +68,98 @@ impl Severity for UbsanWarning {
...
-                    offset: crashline.offset,
-                })
+            let Ok(stack_crashline) = get_crash_line(crashline) else {
+                return header_crashline;
+            };
+            let Ok(header_crashline) = header_crashline else {
+                return Ok(stack_crashline);
+            };
+            let CrashLine::Source(stack_crashline) = stack_crashline else {
+                // header_crashline may be CrashLine::Source
```
Comment (SweetVishnya) [inline libcasr/src/ubsan.rs:138]: may be a

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: libcasr/src/ubsan.rs (hunk 3 above)
Comment (hkctkuy) [reply libcasr/src/ubsan.rs]: Done

File: libcasr/src/ubsan.rs (hunk 4 above)
Comment (hkctkuy) [reply libcasr/src/ubsan.rs]: Done

**Review by SweetVishnya — APPROVED** [review]

**Pull request merged**